docs/plugins.md

```diff
@@ -0,0 +1,150 @@
+# About
+
+Plugins are external components allowing to execute functions on specific event triggering.  
+There is several types of plugins:
+
+* Hooks event: just listen to events and perform other actions (ie: a log plugin). They do not respond anything directly, they just listen.
+* Pipes event: perform an action and return something. Kuzzle is waiting for all pipes event are performed before continuing.
+
+# Configuration
+
+For customize plugins, you can create a file `config/customPlugins.json`. This file can override default plugins file `config/customPlugins.json` for add/remove plugins and their configurations.  
+If you're using docker, you can create your own `customPlugins.json` file and mount it in `/var/app/config/customPlugins.json`. In `docker-compose.yml` file, you can have something like
+
+```
+kuzzle:
+  image: kuzzleio/kuzzle
+  volumes:
+    - "host/path/to/customPlugins.json:/var/app/config/customPlugins.json"
+  ports:
+    - "7512:7512"
+  links:
+    - elasticsearch
+    - redis
+```
+
+A plugin configuration can have attributes:
+
+* `url`: a git url where the plugin can be find and clone.
+* `version`: a version corresponding to the version given in the file `package.json` in the plugin module.
+* `customConfig`: config for the plugin. Each plugin have a different configuration (required or optional), check the corresponding plugin documentation for more information.
+* `defaultConfig`: Don't edit this attribute. The defaultConfig is provided by the plugin itself. If you need to change the configuration, edit the `customConfig` attribute
+
+**Note:** 
+* Url or version are required. The url is checked first, so if you have set a version and an url, the version will be ignored.
+
+# Default plugins
+
+## Logger
+
+By default, the logger plugin is enabled and configured for using the service `winston` (refer to the kuzzle-plugin-logger documentation for more information).  
+
+# How to create a plugin
+
+A plugin is a Javascript module that can be installed with NPM or via a public GIT repository.
+
+## Configuration
+
+The module must have a `package.json` file with a `pluginInfo` entry. The optional `defaultConfig` will be copied in files `config/defaultPlugins.json` and `config/customPlugins.json` in Kuzzle.
```

Comment:
just to be sure.... the config/defaultPlugins.json and config/customPlugins.json are not gitinored (only config/plugins/ folder is).
So these generated files would be commited... is it what we want ?